### PR TITLE
Periodicstlbc

### PR DESCRIPTION
**Pull request by janmees**
lets see if this works

### CODE REVIEW

**Comment by kauzlari** [comment]
AUTHORS "(2015-1016)" -> "(2016)"

**Comment by kauzlari** [comment]
boundary_stl_periodic.h:31:
please change
#ifndef BOUNDARY_STL_PERIODIC_H
#define	BOUNDARY_STL_PERIODIC_H
into
#ifndef __BOUNDARY_STL_PERIODIC_H
#define __BOUNDARY_STL_PERIODIC_H
to preserve name convention

**Comment by kauzlari** [comment]
Is there a Makefile in the repository? If yes, please remove. This should be created by the autotools and not be part of the repository. But Makefile.am should stay!

**Comment by kauzlari** [comment]
boundary_stl_periodic.cpp: 164, 167, 179, 182:
These lines contain comparisons of doubles representing coordinates. Obviously, with your test-input they worked so far. But could you please perform a "sensitivity" test. I.e., what happens, if there is a slight deviation in the last digit for two supposedly identical coordinates. If necessary create this deviation "by hand" in the STL-file. I fear, for a safe operation, we must introduce an "epsilon". This could be one of the epsilons already existing in the code (should be somewhere in the 'geometry' directory). Or can we expect the user to deliver STL-files where this is absolutely not necessary, by making a statement such as "STL-files produced by CAD software XYZ, are guaranteed to be read by SYMPLER without difficulties."? Then, XYZ would have to be free, non-commercial software. Btw, did you test both ASCII and binary files?

**Comment by kauzlari** [comment]
MSG_DEBUG("Boundary Stl Periodic", "Setup completed.");
->
MSG_DEBUG("BoundaryStlPeriodic::setup", "Setup completed.");
Please check also for all other cases that this format is used.

**Comment by kauzlari** [comment]
wall.h: I think your idea was that findcorner(int i) implements default behaviour which is overwritten by child classes. Wouldn't it be better to make the function in wall.h pure virtual? As far as I remember, class Wall is not meant to be instantiated and the code does not do that anywhere.

**Comment by janmees** [comment]
The extended initializer comment is the warning i got while compiling when i used {0,0,0}. Doing it old fashion way x=0; No warning so i changed  the way it was implemented since i wasnt sure if sympler is meant to require c++11. Ill look at the rest as soon as possible.

**Comment by kauzlari** [comment]
Please add an example to the "examples" folder that does exactly the same thing as
[https://github.com/kauzlari/sympler/blob/master/examples/SPHperiodicPoiseuille.xml](url)
and hence produces exactly the same results. Then we can merge.

File: examples/SPHSTLperiodicPoiseuille.xml [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  <FParticleVels
+      species = "fluid"
+      expression = "uVecY(0.005)"
+      />
+  
+  <Phase
+      smartCells = "0"
+      >
+    <LinkedListCreator/>
+
```
Comment (kauzlari) [inline examples/SPHSTLperiodicPoiseuille.xml]: We need some changes/additions in the following comment:
- Note that I originally mentioned the required directories to be created beforehand in the comment on the very top of the XML-file (which I would put back there). Either you should say how "PATH/" is to be understood (e.g., user still has to adjust this to his/her needs) or you should remove "PATH/" and assume the STL-file to reside in the directory where sympler is called and say this briefly here. 
- The user should be told the dimensions of the box which will result from the settings here and from the used STL-file
- We should add s.th. like: "The aspect ratio of the cuboid can be changed by the scaleX/Y/Z attributes."
- The current explanation of why the dimensions are exactly as they are should be nicely merged with the new information. Note that this is an explanation for the special case simulated here. So take care where to place it relative to the rest. Either go in the whole comment from special to general or the other way round, but do not intermingle the two.

**Pull request merged**